ipaserver/install/gcinstance.py

```diff
@@ -672,23 +698,45 @@ def __configure_oc_filter_rewriter(self):
         )
 
     def __add_service_alias(self):
-        # We share principal with the primary ldap service
-        principal = unicode(Principal(
-            (self.service_prefix, self.fqdn), realm=self.realm))
-        api.Command.service_add_principal(principal, self.principal)
+        for (primary, alias) in DC_ALIASES:
+            principal = unicode(Principal(
+                ipautil.template_str(primary, self.sub_dict),
+                realm=self.realm))
+            principal_alias = unicode(Principal(
+                ipautil.template_str(alias, self.sub_dict),
+                realm=self.realm))
+            api.Command.service_add_principal(
+                principal, principal_alias)
 
     def __remove_service_alias(self):
-        # We share principal with the primary ldap service
-        principal = unicode(Principal(
-            (self.service_prefix, api.env.host),
-            realm=api.env.realm))
-        principal_alias = unicode(Principal(
-            (self.service_prefix, api.env.host, api.env.domain),
-            realm=api.env.realm))
+        # self.sub_dict is not populated, so we have to derive values
+        # from a global trust configuration
         try:
-            api.Command.service_remove_principal(principal, principal_alias)
-        except (errors.AttrValueNotFound, errors.NotFound):
-            pass
+            trustconfig = api.Command.trustconfig_show()['result']
+        except errors.NotFound:
+            # No trust configuration anymore, it is real uninstall
+            # Skip the removal
+            return
+
+        sub_dict = dict(
+            FQDN=api.env.host,
+            REALM=api.env.realm,
+            DOMAIN=api.env.domain,
+            DOMAINGUID=trustconfig.get('ipantdomainguid')[0],
```

Comment:
With this code, the install creates a krbprincipalname `E3514235-4B06-11D1-AB04-00C04FC2DCD2/xf2HionxQKSOLwce75aU9w==/domain.com@DOMAIN.COM`. Not sure whether the value `xf2HionxQKSOLwce75aU9w==` would cause an issue but I was expecting a readable GUID.

Reply:
right, so we need to unparse it